### ISSUE TITLE
Write test codes

### ISSUE THREAD

**Issue by dlguswo333** [dlguswo333/react-window-infinite-scroll#2]
Write test codes for package's reliabilty.
Use headless browser tool if possible.

**Comment by dlguswo333** [comment]
Things have been working great until I met the problem:
```tsx
test('Load more data with some initial data', async () => {
  const Component = () => {
    const [data, setData] = useState(['0']);
    const outerRef = useRef<HTMLElement>(null);
    const loadMoreItems = async () => {
      setData(data.concat(data.length.toString()));
    };
    const isItemsLoaded = (index: number) => {
      const ret = !(index === data.length && index < 3);
      return ret;
    };
    const itemCount = data.length + (data.length < 5 ? 1 : 0);

    const Row = ({index, style}: {index: number, style: CSSProperties}) => (
      <div style={style}>{data[index]}</div>
    );

    return <InfiniteScroll
      data={data}
      loadMoreItems={loadMoreItems}
      isItemLoaded={isItemsLoaded}
      itemCount={itemCount}
      threshold={1}
      outerRef={outerRef}
      scrollOffset={30}
    >
      {({onItemsRendered}) => <FixedSizeList
        height={300}
        itemCount={itemCount}
        width='100%'
        onItemsRendered={onItemsRendered}
        itemSize={30}
        outerRef={outerRef}
        className='Outer'
      >
        {Row}
      </FixedSizeList>}
    </InfiniteScroll>;
  };

  render(<Component />);
  expect(screen.getByText('0')).toBeTruthy();
  await waitFor(() => expect(screen.getByText('0')).toBeTruthy());
  await waitFor(() => expect(screen.getByText('1')).toBeTruthy());
  await waitFor(() => expect(screen.getByText('2')).toBeTruthy()); // Error!
});
```
This test case throws an error saying that `'2'` is not rendered.

In short, before `loadMoreItems` is being resolved, `-loadMoreItems` is called.
However `pending.current` is not set to `false` because `loadMoreItems` has not yet been resolved,
thus it will not load more items.
If I add `await` keyword before `getOuterElement()`, test goes fine.
Thus I presume this seems to be an issue related to jest's internal clock.

https://github.com/dlguswo333/react-window-infinite-scroll/blob/62576d86e4e8996997437fb4d5ab9acce386e9a4/package/src/InfiniteScroll.tsx#L69-L73

I don't know how jest works, but as long as jest fails and browser succeeds, I will follow as browser says.
Maybe I need to switch to headless browser tools; such as playwright.

**Comment by dlguswo333** [comment]
Even if it may be the case where the promise resolves slower than the data changes,
I still presume it is very rare cases.
The case I will deal with is where `loadMoreItems` synchronously runs.
Allow `loadMoreItems` to return that is not a promise. **await** only when the return values are `promise`s.

**Comment by dlguswo333** [comment]
How to scroll elements in playwright: https://github.com/microsoft/playwright/discussions/11547

**Comment by dlguswo333** [comment]
Playwright runs test codes on NodeJS and page codes on browser.

>Since Playwright runs in Node.js, you can debug it with your debugger of choice e.g. using console.log or inside your IDE or directly in VS Code ...
>https://playwright.dev/docs/running-tests

>Playwright scripts run in your Playwright environment. Your page scripts run in the browser page environment. Those environments don't intersect, they are running in different virtual machines in different processes and even potentially on different computers.
>https://playwright.dev/docs/evaluating

It means you cannot share JS contexts and variables directly. Even if you can pass from playwright context to browser code, their links will be perished.

Instead you can run `evaluate`, where you can pass a function get the return value.

>The [page.evaluate()](https://playwright.dev/docs/api/class-page#page-evaluate) API can run a JavaScript function in the context of the web page and bring results back to the Playwright environment. Browser globals like window and document can be used in evaluate.

```js
const href = await page.evaluate(() => document.location.href);
```

However, I doubt it may retrieve non-primitive data, such as functions or streams.
But such cases are not possible even outside NodeJS worlds and extremely rare in real world.

**Comment by dlguswo333** [comment]
Closes in favor of PR #16 .
